### PR TITLE
Refactor fideslog to more closely match fides api structure

### PR DESCRIPTION
**Pull request by TheAndrewJackson**
This PR updates fideslog to mostly match the fides api structure. This should make it easier for developers coming from fides pick things up quicker.

### CODE REVIEW

File: fideslog/api/routes/health.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
 from fastapi import APIRouter, Request, status
 from fastapi.responses import JSONResponse
 
-from ..config import config
-from ..endpoints.events import router as event_router
+from fideslog.api.config import config
 
-api_router = APIRouter()
-api_router.include_router(event_router)
+health_router = APIRouter()
```
Comment (PSalant726) [inline fideslog/api/routes/health.py:6]: Now that there's a dedicated `health_router`, we can move the `tags` from the endpoint definition on L56 to this `APIRouter()` invocation.

File: fideslog/api/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .models import AnalyticsEvent
```
Comment (PSalant726) [inline fideslog/api/models/__init__.py:1]: I'm not getting any errors with this removed.
```suggestion
```

File: fideslog/api/routes/events.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+from logging import getLogger
+
+from fastapi import APIRouter, Depends, HTTPException, Request, status
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy.orm import Session
+
+from fideslog.api.config import config
+from fideslog.api.database.data_access import create_event
+from fideslog.api.database.database import get_db
+from fideslog.api.schemas.analytics_event import AnalyticsEvent
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:10]: Changing these to relative imports is working just fine for me:
```suggestion
from ..config import config
from ..database.data_access import create_event
from ..database.database import get_db
from ..schemas.analytics_event import AnalyticsEvent
```

File: fideslog/api/routes/events.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                "X-RateLimit-Remaining": {
+                    "description": "The number of remaining requests in the current period.",
+                    "schema": {"type": "integer"},
+                },
+                "X-RateLimit-Reset": {
+                    "description": "The number of seconds left in the current period.",
+                    "schema": {"type": "integer"},
+                },
+            },
+        }
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:52]: This is rewritten in 3 places. It would be nice to DRY it up by defining a constant in an `api/errors.py` file, like:
```py
# api/errors.py

from logging import getLogger
from typing import Dict, Union

from fastapi import HTTPException, status

from .config import config

log = getLogger(__name__)


class TooManyRequestsError(HTTPException):
    """
    To be raised when a request exceeds the configured rate limit.
    """

    MESSAGE = "Rate limit exceeded"

    def __init__(self, error: Exception) -> None:
        log.error("%s: %s", self.MESSAGE, error, exc_info=error)
        super().__init__(
            status.HTTP_429_TOO_MANY_REQUESTS,
            detail={"message": self.MESSAGE, "limit": config.server.request_rate_limit},
        )

    @classmethod
    def doc(cls) -> Dict[str, Union[Dict, str]]:
        """
        Returns the documentation for a 429 Too Many Requests response,
        in the OpenAPI spec format.
        """

        return {
            "content": {
                "application/json": {
                    "example": {
                        "error": f"Rate limit exceeded: {config.server.request_rate_limit}"
                    },
                    "schema": {
                        "type": "object",
                        "properties": {"error": {"type": "string"}},
                    },
                }
            },
            "description": "Rate limit exceeded",
            "headers": {
                "Retry-After": {
                    "description": "The datetime after which to retry the request.",
                    "schema": {"type": "http-date"},
                },
                "X-RateLimit-Limit": {
                    "description": "The number of allowed requests in the current period.",
                    "schema": {"type": "integer"},
                },
                "X-RateLimit-Remaining": {
                    "description": "The number of remaining requests in the current period.",
                    "schema": {"type": "integer"},
                },
                "X-RateLimit-Reset": {
                    "description": "The number of seconds left in the current period.",
                    "schema": {"type": "integer"},
                },
            },
        }

```
Then, here + everywhere else this is done, you can instead just do:
```py
from ..errors import TooManyRequestsError
```
```suggestion
        status.HTTP_429_TOO_MANY_REQUESTS: TooManyRequestsError.doc()
```

File: fideslog/api/routes/events.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    },
+    status_code=status.HTTP_201_CREATED,
+)
+async def create(
+    request: Request,  # pylint: disable=unused-argument
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:57]:
```suggestion
    _: Request,
```

File: fideslog/api/routes/events.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    """
+
+    try:
+        create_event(database=database, event=event)
+    except DBAPIError as err:
+        log.error("Failed to create event: %s", err, exc_info=True)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Failed to create event",
+        ) from None
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:72]: This code defines a `500 Internal Server Error` as a possible response, but that response is not documented in the `responses` dict on L21. Similar to my above comment, we should define an `InternalServerError` class in `api/errors.py`, and simply raise a new instance of it here + document the response type like:
```py
# L21

responses={
  ...,
  status.HTTP_500_INTERNAL_SERVER_ERROR: InternalServerError.doc(),
}
```

File: fideslog/api/routes/events.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    "description": "The number of remaining requests in the current period.",
+                    "schema": {"type": "integer"},
+                },
+                "X-RateLimit-Reset": {
+                    "description": "The number of seconds left in the current period.",
+                    "schema": {"type": "integer"},
+                },
+            },
+        }
+    },
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:114]: After addressing my above comments, this can become:
```suggestion
        status.HTTP_429_TOO_MANY_REQUESTS: TooManyRequestsError.doc(),
        status.HTTP_500_INTERNAL_SERVER_ERROR: InternalServerError.doc(),
```

File: fideslog/api/routes/events.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    """
+
+    try:
+        create_event(database=database, event=event)
+    except DBAPIError as err:
+        log.error("Failed to create event: %s", err, exc_info=True)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Failed to create event",
+        ) from None
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:133]: After addressing my above comments, this can just be:
```suggestion
        raise InternalServerError(...) from err
```

**Review by PSalant726 — APPROVED** [review]
Nothing blocking here - this is a nice cleanup! Just a few suggestions to clean things up even more.

**Comment by TheAndrewJackson** [comment]
@PSalant726 Great feedback! I like the pattern of storing the response docs on the exception itself. That pattern should be ported back to fides. Everything should be good to go now 🚀

File: fideslog/api/routes/events.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    database: Session = Depends(get_db),
+) -> AnalyticsEvent:
+    """
+    Create a new analytics event.
+    """
+
+    try:
+        create_event(database=database, event=event)
+    except DBAPIError as err:
+        raise InternalServerError(...) from err  # type: ignore
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:38]: Ha, we can't literally do `(...)` here - it's invalid. I was just trying to give you a shorthand example. 
```suggestion
        raise InternalServerError(err) from err
```

File: fideslog/api/routes/events.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+    return event
```
Comment (PSalant726) [inline fideslog/api/routes/events.py:67]: I didn't notice this earlier, but I think you're mixing in changes from the other branch you're working on? This is a new endpoint and probably belongs on it's own PR.

**Review by PSalant726 — CHANGES_REQUESTED** [review]

File: fideslog/api/routes/events.py (hunk 9 above)
Comment (TheAndrewJackson) [reply fideslog/api/routes/events.py:38]: I thought you knew something I didn't 😅  That makes way more sense. I'll fix that in both of them

File: fideslog/api/routes/events.py (hunk 10 above)
Comment (TheAndrewJackson) [reply fideslog/api/routes/events.py:67]: Good catch. I'll remove it from this PR.

**Review by PSalant726 — APPROVED** [review]
